new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTraefikLoadBalancer.java

```diff
@@ -0,0 +1,353 @@
+
+      bindDomainName(domainUid);
+    }
+
+    // verify load balancing works when 2 domains are running in the same namespace
+    for (String domainUid : domains) {
+      verifyLoadbalancing(domainUid, replicaCount, managedServerNameBase);
+    }
+  }
+
+  private void verifyLoadbalancing(String domainUid, int replicaCount, String managedServerNameBase) {
+    //access application in managed servers through traefik load balancer
+    logger.info("Accessing the clusterview app through traefik load balancer");
+    String curlRequest = String.format("curl --silent --show-error --noproxy '*' "
+        + "-H 'host: %s' http://%s:%s/clusterview/ClusterViewServlet",
+        domainUid + "." + domainNamespace + "." + "cluster-1" + ".test", K8S_NODEPORT_HOST, nodeportshttp);
+    List<String> managedServers = new ArrayList<>();
+    for (int i = 1; i <= replicaCount; i++) {
+      managedServers.add(managedServerNameBase + i);
+    }
+    assertThat(verifyClusterMemberCommunication(curlRequest, managedServers, 20))
+        .as("Verify applications from cluster can be acessed through the traefik loadbalancer.")
+        .withFailMessage("application not accessible through traefik loadbalancer.")
+        .isTrue();
+
+    boolean hostRouting = false;
+    //access application in managed servers through traefik load balancer and bind domain in the JNDI tree
+    logger.info("Accessing the clusterview app through traefik load balancer");
+    String curlCmd = String.format("curl --silent --show-error --noproxy '*' "
+        + "-H 'host: %s' http://%s:%s/clusterview/ClusterViewServlet?domainTest=" + domainUid,
+        domainUid + "." + domainNamespace + "." + "cluster-1" + ".test", K8S_NODEPORT_HOST, nodeportshttp);
+
+    // call the webapp and verify the bound domain name to determine
+    // the requests are sent to the correct cluster members.
+    for (int i = 0; i < 10; i++) {
+      ExecResult result;
+      try {
+        result = ExecCommand.exec(curlCmd, true);
+        String response = result.stdout().trim();
+        if (response.contains(domainUid)) {
+          hostRouting = true;
+        }
+        logger.info("Response for iteration {0}: exitValue {1}, stdout {2}, stderr {3}",
+            i, result.exitValue(), response, result.stderr());
+      } catch (IOException | InterruptedException ex) {
+        //
+      }
+    }
+    assertTrue(hostRouting, "Host routing is not working");
+
+  }
+
+  private void bindDomainName(String domainUid) {
+    //access application in managed servers through traefik load balancer and bind domain in the JNDI tree
+    logger.info("Accessing the clusterview app through traefik load balancer");
+    String curlCmd = String.format("curl --silent --show-error --noproxy '*' "
+        + "-H 'host: %s' http://%s:%s/clusterview/ClusterViewServlet?bindDomain=" + domainUid,
+        domainUid + "." + domainNamespace + "." + "cluster-1" + ".test", K8S_NODEPORT_HOST, nodeportshttp);
```

Comment:
You can change this to: domainUid + "." + domainNamespace + ".cluster-1.test", K8S_NODEPORT_HOST, nodeportshttp);

Reply:
fixed